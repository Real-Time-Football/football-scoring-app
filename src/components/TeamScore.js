import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { parseTeamLogo, parseTeamName } from './matchHelpers'

const DefaultTeamContainer = styled.article`
    display: flex;
    flex-direction: ${props => (props.reverse ? 'row-reverse' : 'row')};
    align-items: center;
`

const DefaultTeamLogo = styled.img`
    height: 60px;
    width: 60px;
`

const DefaultTeamName = styled.header`
    font-size: 20px;
    margin: ${props => (props.reverse ? '0 0 0 20px' : '0 20px 0 0')};
`

const DefualtScore = styled.div`
    font-size: 38px;
    margin: ${props => (props.reverse ? '0 40px 0 10px' : '0 10px 0 40px')};
`

const TeamScore = ({ TeamContainer, TeamName, TeamLogo, Score, team, score = 0, side }) => (
    <TeamContainer data-side={side} reverse={side === 'VISITORS'}>
        <TeamName reverse={side === 'VISITORS'}>
            <h1>{parseTeamName(team)}</h1>
        </TeamName>
        <aside>
            <TeamLogo className="team-logo" src={parseTeamLogo(team)} />
        </aside>
        <Score reverse={side === 'VISITORS'}>
            <strong>{score}</strong>
        </Score>
    </TeamContainer>
)

TeamScore.propTypes = {
    TeamContainer: PropTypes.elementType,
    TeamName: PropTypes.elementType,
    TeamLogo: PropTypes.elementType,
    Score: PropTypes.elementType,
    team: PropTypes.string.isRequired,
    score: PropTypes.number,
    side: PropTypes.string.isRequired,
    Logo: PropTypes.elementType,
}

TeamScore.defaultProps = {
    TeamContainer: DefaultTeamContainer,
    TeamName: DefaultTeamName,
    TeamLogo: DefaultTeamLogo,
    Score: DefualtScore,
}

export default TeamScore
