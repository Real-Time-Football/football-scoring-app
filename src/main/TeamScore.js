import React from 'react'
import PropTypes from 'prop-types'

const TeamFlag = ({ teamDescription, score, ...rest }) => (
    <div {...rest}>
        <div>{teamDescription}</div>
        <div>{score}</div>
    </div>
)

TeamFlag.propTypes = {
    teamDescription: PropTypes.string.isRequired,
    score: PropTypes.number,
}

export default TeamFlag
