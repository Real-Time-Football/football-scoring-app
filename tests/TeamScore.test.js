import React from 'react'
import { render, screen } from '@testing-library/react'
import TeamScore from '../src/components/TeamScore'

describe('Team Score', () => {
    // let wrapper
    let component
    const team = 'SÃ£o Paulo'
    const side = 'HOME'

    beforeEach(() => {
        component = render(<TeamScore team={team} side={side} />)
    })

    it('passes `team` through to the <header>', () => {
        expect(component.getByTestId('team-name-header').textContent).toBe(team)
    })

    it('passes `score` through to the <div>', () => {
        const score = 1
        component.rerender(<TeamScore team={team} side={side} score={score} />)
        expect(component.getByTestId('score-label').textContent).toBe('1')
    })

    it('has a initial `score` of 0', () => {
        expect(component.getByTestId('score-label').textContent).toBe('0')
    })

    it('render a <image> with team logo', () => {
        const imgUrl = 'src/assets/images/logo-sao-paulo-256.png'
        expect(screen.getByRole('img').getAttribute('src')).toBe(imgUrl)
    })

    it('renders correctly', () => {
        expect(component).toMatchSnapshot()
    })
})
