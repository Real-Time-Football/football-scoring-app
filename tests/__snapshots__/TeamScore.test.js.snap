// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Score renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  height: 60px;
  width: 60px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-size: 20px;
  margin: 0 20px 0 0;
}

.c3 {
  font-size: 38px;
  margin: 0 10px 0 40px;
}

<body>
    <div>
      <article
        class="c0"
        data-side="HOME"
      >
        <header
          class="c1"
        >
          <h1
            data-testid="team-name-header"
          >
            São Paulo
          </h1>
        </header>
        <aside>
          <img
            class="c2 team-logo"
            src="src/assets/images/logo-sao-paulo-256.png"
          />
        </aside>
        <div
          class="c3"
        >
          <strong
            data-testid="score-label"
          >
            0
          </strong>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="TeamScore__TeamContainer-sc-1h0nwh1-1 dTLVJC"
      data-side="HOME"
    >
      <header
        class="TeamScore__TeamName-sc-1h0nwh1-2 HdRNo"
      >
        <h1
          data-testid="team-name-header"
        >
          São Paulo
        </h1>
      </header>
      <aside>
        <img
          class="TeamScore__TeamLogo-sc-1h0nwh1-0 hrrmtp team-logo"
          src="src/assets/images/logo-sao-paulo-256.png"
        />
      </aside>
      <div
        class="TeamScore__Score-sc-1h0nwh1-3 jVzBBJ"
      >
        <strong
          data-testid="score-label"
        >
          0
        </strong>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
